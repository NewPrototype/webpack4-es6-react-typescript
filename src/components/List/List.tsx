import * as React from 'react';
import './List.less';

import { Timeline,Alert} from 'antd';




// import { withRouter } from 'react-router-dom'


interface Props {

}


const status = {
  "start": '开始1',
  "end": '结束'
}

class List extends React.Component<Props, { text: string }> {
  constructor(props: Props) {
    super(props);
    this.state = {
      text: status.start
    };
  }
  onClick = () => {
    this.setState({
      text: this.state.text == status.start ? status.end : status.start
    })
  }

  componentWillMount() {

  }

  componentDidMount() { }

  componentWillUnmount() { }


  render() {
    return (
      <div className="List" onClick={this.onClick} >
        <div >
                {this.state.text}
        </div>  
        <Alert message="Success Text" type="success" />,
        <Timeline>
    <Timeline.Item>Create a services site 2015-09-01</Timeline.Item>
    <Timeline.Item>Solve initial network problems 2015-09-01</Timeline.Item>
    <Timeline.Item>Technical testing 2015-09-01</Timeline.Item>
    <Timeline.Item>Network problems being solved 2015-09-01</Timeline.Item>
  </Timeline>,
      </div>
    );
  }
}

export default List