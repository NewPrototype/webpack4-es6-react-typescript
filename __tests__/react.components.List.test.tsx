import * as React from 'react';
import List from './../src/components/List';


import { create } from 'react-test-renderer';


describe('React Components', () => {
  const component = create(
    <List />
  );
  let tree = component.toJSON();
  it('List init render ', () => {
    expect(tree).toMatchSnapshot()
  })
  it('List onClick render 第一次点击', () => {
    tree.props.onClick();
    tree = component.toJSON();
    expect(tree).toMatchSnapshot()
  })
  it('List onClick render 第二次点击', () => {
    tree.props.onClick();
    tree = component.toJSON();
    expect(tree).toMatchSnapshot()
  })
})
