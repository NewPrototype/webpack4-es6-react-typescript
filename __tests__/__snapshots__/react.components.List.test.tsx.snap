// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React Components List init render  1`] = `
<div
  className="List"
  onClick={[Function]}
>
  <div>
    开始1
  </div>
  <div
    className="ant-alert ant-alert-success ant-alert-no-icon"
    data-show={true}
  >
    <span
      className="ant-alert-message"
    >
      Success Text
    </span>
    <span
      className="ant-alert-description"
    />
  </div>
  ,
  <ul
    className="ant-timeline"
  >
    <li
      className="ant-timeline-item"
    >
      <div
        className="ant-timeline-item-tail"
      />
      <div
        className="ant-timeline-item-head ant-timeline-item-head-blue"
        style={
          Object {
            "borderColor": undefined,
          }
        }
      />
      <div
        className="ant-timeline-item-content"
      >
        Create a services site 2015-09-01
      </div>
    </li>
    <li
      className="ant-timeline-item"
    >
      <div
        className="ant-timeline-item-tail"
      />
      <div
        className="ant-timeline-item-head ant-timeline-item-head-blue"
        style={
          Object {
            "borderColor": undefined,
          }
        }
      />
      <div
        className="ant-timeline-item-content"
      >
        Solve initial network problems 2015-09-01
      </div>
    </li>
    <li
      className="ant-timeline-item"
    >
      <div
        className="ant-timeline-item-tail"
      />
      <div
        className="ant-timeline-item-head ant-timeline-item-head-blue"
        style={
          Object {
            "borderColor": undefined,
          }
        }
      />
      <div
        className="ant-timeline-item-content"
      >
        Technical testing 2015-09-01
      </div>
    </li>
    <li
      className="ant-timeline-item ant-timeline-item-last"
    >
      <div
        className="ant-timeline-item-tail"
      />
      <div
        className="ant-timeline-item-head ant-timeline-item-head-blue"
        style={
          Object {
            "borderColor": undefined,
          }
        }
      />
      <div
        className="ant-timeline-item-content"
      >
        Network problems being solved 2015-09-01
      </div>
    </li>
  </ul>
  ,
</div>
`;

exports[`React Components List onClick render 第一次点击 1`] = `
<div
  className="List"
  onClick={[Function]}
>
  <div>
    结束
  </div>
  <div
    className="ant-alert ant-alert-success ant-alert-no-icon"
    data-show={true}
  >
    <span
      className="ant-alert-message"
    >
      Success Text
    </span>
    <span
      className="ant-alert-description"
    />
  </div>
  ,
  <ul
    className="ant-timeline"
  >
    <li
      className="ant-timeline-item"
    >
      <div
        className="ant-timeline-item-tail"
      />
      <div
        className="ant-timeline-item-head ant-timeline-item-head-blue"
        style={
          Object {
            "borderColor": undefined,
          }
        }
      />
      <div
        className="ant-timeline-item-content"
      >
        Create a services site 2015-09-01
      </div>
    </li>
    <li
      className="ant-timeline-item"
    >
      <div
        className="ant-timeline-item-tail"
      />
      <div
        className="ant-timeline-item-head ant-timeline-item-head-blue"
        style={
          Object {
            "borderColor": undefined,
          }
        }
      />
      <div
        className="ant-timeline-item-content"
      >
        Solve initial network problems 2015-09-01
      </div>
    </li>
    <li
      className="ant-timeline-item"
    >
      <div
        className="ant-timeline-item-tail"
      />
      <div
        className="ant-timeline-item-head ant-timeline-item-head-blue"
        style={
          Object {
            "borderColor": undefined,
          }
        }
      />
      <div
        className="ant-timeline-item-content"
      >
        Technical testing 2015-09-01
      </div>
    </li>
    <li
      className="ant-timeline-item ant-timeline-item-last"
    >
      <div
        className="ant-timeline-item-tail"
      />
      <div
        className="ant-timeline-item-head ant-timeline-item-head-blue"
        style={
          Object {
            "borderColor": undefined,
          }
        }
      />
      <div
        className="ant-timeline-item-content"
      >
        Network problems being solved 2015-09-01
      </div>
    </li>
  </ul>
  ,
</div>
`;

exports[`React Components List onClick render 第二次点击 1`] = `
<div
  className="List"
  onClick={[Function]}
>
  <div>
    开始1
  </div>
  <div
    className="ant-alert ant-alert-success ant-alert-no-icon"
    data-show={true}
  >
    <span
      className="ant-alert-message"
    >
      Success Text
    </span>
    <span
      className="ant-alert-description"
    />
  </div>
  ,
  <ul
    className="ant-timeline"
  >
    <li
      className="ant-timeline-item"
    >
      <div
        className="ant-timeline-item-tail"
      />
      <div
        className="ant-timeline-item-head ant-timeline-item-head-blue"
        style={
          Object {
            "borderColor": undefined,
          }
        }
      />
      <div
        className="ant-timeline-item-content"
      >
        Create a services site 2015-09-01
      </div>
    </li>
    <li
      className="ant-timeline-item"
    >
      <div
        className="ant-timeline-item-tail"
      />
      <div
        className="ant-timeline-item-head ant-timeline-item-head-blue"
        style={
          Object {
            "borderColor": undefined,
          }
        }
      />
      <div
        className="ant-timeline-item-content"
      >
        Solve initial network problems 2015-09-01
      </div>
    </li>
    <li
      className="ant-timeline-item"
    >
      <div
        className="ant-timeline-item-tail"
      />
      <div
        className="ant-timeline-item-head ant-timeline-item-head-blue"
        style={
          Object {
            "borderColor": undefined,
          }
        }
      />
      <div
        className="ant-timeline-item-content"
      >
        Technical testing 2015-09-01
      </div>
    </li>
    <li
      className="ant-timeline-item ant-timeline-item-last"
    >
      <div
        className="ant-timeline-item-tail"
      />
      <div
        className="ant-timeline-item-head ant-timeline-item-head-blue"
        style={
          Object {
            "borderColor": undefined,
          }
        }
      />
      <div
        className="ant-timeline-item-content"
      >
        Network problems being solved 2015-09-01
      </div>
    </li>
  </ul>
  ,
</div>
`;
